doc/dev/release_checklist.md

```diff
@@ -132,6 +144,7 @@ We'll use `v4.7.0-rc1` as the intended release version in this example.
 
 - Decide which nightly release you want to turn into a release candidate.
   We will use `nightly-2024-02-29` in this example.
+- It is essential to choose the nightly that will become the release candidate as early as possible, to avoid confusion.
```

Comment:
Maybe point out somewhere around here that the same checklist script can now be used with the `v4.7.0-rc1` argument.